sim/rogue/talents.go

```diff
@@ -440,8 +443,15 @@ func (rogue *Rogue) registerBladeFlurryCD() {
 			// Undo armor reduction to get the raw damage value.
 			curDmg = result.Damage / result.ResistanceMultiplier
 
-			bfHit.Cast(sim, rogue.Env.NextTargetUnit(result.Target))
-			bfHit.SpellMetrics[result.Target.UnitIndex].Casts--
+			curTarget := rogue.Env.NextTargetUnit(result.Target)
+			for i := int32(0); i < rogue.bladeFlurryTargetCount; i++ {
```

Comment:
If we do this without checking the number of targets the user actually has selected it will loop back around to targets you've already hit. I would assign a variable above the aura registration
```
numTargets := rogue.Env.GetNumTargets()
```
then in here, take the min of that and the `bladeFlurryTargetCount`
```
for i := int32(0); i < min(numTargets, rogue.bladeFlurryTargetCount; i++ {
```

Reply:
Oh I was basing the loop on the cleave interaction which has this logic. The loop is broken if we circle back to the original target of the hit
```
				if curTarget == result.Target {
					break
				}
```

Reply:
Oh that works too! It's probably not quite as graceful but if you just copied another example that's ok too

Reply:
Oh looking at the Cleave code, I'm pretty sure I was trying to refactor it but then forgot to delete the target comparison